workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/DetailsStore.cs

```diff
@@ -81,16 +82,22 @@ public DetailsStore(SchemaBundle bundle, List<string> serializationOverrides, IF
                 .Select(file => file.CanonicalPath)
                 .ToList().AsReadOnly();
 
-            foreach (var kv in Types)
-            {
-                kv.Value.PopulateFields(this);
-                kv.Value.PopulateChildren(this);
-            }
-
-            foreach (var kv in Components)
-            {
-                kv.Value.PopulateFields(this);
-            }
+            Types = new Dictionary<string, UnityTypeDetails>(types
+                .Select(kv =>
+                {
+                    kv.Value.PopulateFields(this);
+                    kv.Value.PopulateChildren(this);
+                    return kv;
+                })
+                .Where(kv => kv.Value.IsValid()));
```

Comment:
Shouldn't we be filtering out the offending field in the type rather than the whole type? 

Imagine I had the following schema: 

```
type MyType {
	int32 data = 1;
}
```

And this type was used all over my codebase. Then at some point in the future someone adds an inner type and field (that clashes):

```
type MyType {
	type Inner {}
	
	int32 data = 1;
	Inner inner = 2;
}
```

Now I'll get code compilation errors since `MyType` has disappeared entirely rather than the new field being omitted? 

It also aligns with us dropping the event/command when it clashes instead of the entire component